hbase-shell/src/main/ruby/hbase/admin.rb

```diff
@@ -1453,6 +1453,97 @@ def list_decommissioned_regionservers
       @admin.listDecommissionedRegionServers
     end
 
+    #----------------------------------------------------------------------------------------------
+    # Retrieve SlowLog Responses from RegionServers
+    def get_slowlog_responses(server_names, args)
+      unless server_names.is_a?(Array) || server_names.is_a?(String)
+        raise(ArgumentError,
+              "#{server_names.class} of #{server_names.inspect} is not of Array/String type")
+      end
+      if server_names == '*'
+        server_names = getServerNames([], true)
+      else
+        server_names_list = get_server_names(server_names)
+        server_names = getServerNames(server_names_list, false)
+      end
+      slow_log_responses = @admin.getSlowLogResponses(server_names)
+      filtered_log_responses = filter_slow_responses(args,
+                                                     slow_log_responses)
+      puts 'Retrieved SlowLog Responses from RegionServers.....'
+      if args.empty?
+        puts slow_log_responses.to_s
+      else
+        puts filtered_log_responses.to_s
+      end
+    end
+
+    def filter_slow_responses(args, slow_log_responses)
+      filtered_log_responses = java.util.ArrayList.new
```

Comment:
Oh, here's where your filters get applied. Please perform filtering on the server-side, where it can be distributed across all the region servers involved, and avoid the network and client overhead.

Reply:
Just re-iterating the comment on Jira to better align it here:
```
Copying Andrew's earlier comment:

"Filtering or sorting on other attributes would be done locally in the client. I think it best to let the client index the list of ResponseDetail however it likes.
The shell commands are one client of the admin APIs. This seems a good place to put additional convenience filtering."

Although I agree with scalability concern but I believe filter done by client(shell) might not be too much of an overhead right? 
```
Even for quite hugh cluster, shell should be able to process it without trouble. 
Please suggest.

FYI @apurtell

Reply:
See my comment on JIRA.